spring-boot/src/main/java/org/springframework/boot/jdbc/DatabaseDriver.java

```diff
@@ -44,19 +45,19 @@
 	 */
 	DERBY("Apache Derby", "org.apache.derby.jdbc.EmbeddedDriver",
 			"org.apache.derby.jdbc.EmbeddedXADataSource",
-			"SELECT 1 FROM SYSIBM.SYSDUMMY1"),
+			"SELECT 1 FROM SYSIBM.SYSDUMMY1", "DERBY"),
```

Comment:
Ah crap, ok.  Now I get the name thing (I should have read everything)

I am not sure (yet) that we should add yet another attribute to that thing. Such soft link is not very interesting anyway. And there's only one user of it so I'd keep that mapping where the `Database` is determined.